pkg/sql/codegen/pai/cluster_config.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pai
+
+import "fmt"
+
+// PSConfig implicates Parameter Server Config
+type PSConfig struct {
+	Count int `json:"count"`
+	GPU   int `json:"gpu"`
+	CPU   int `json:"cpu"`
+}
+
+// WorkerConfig implicates Worker Config
+type WorkerConfig struct {
+	Count int `json:"count"`
+	GPU   int `json:"gpu"`
+	CPU   int `json:"cpu"`
+}
+
+// ClusterConfig implicates PAI distributed task meta
+type ClusterConfig struct {
+	PS     PSConfig     `json:"ps"`
+	Worker WorkerConfig `json:"worker"`
+}
+
+// GetClusterConfig returns ClusterConfig object comes from WITH clause
+func GetClusterConfig(attrs map[string]interface{}) (*ClusterConfig, error) {
```

Comment:
From its content, the type ClusterConfig seems JobConfig?

Reply:
Maybe the name `JobConfig` is a little bit generic, a job may also contains job name, job type etc.